### PR TITLE
Refactored the screens to use the new Screen component.

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

[comment]: <> (Description of the feature or fix)
[comment]: <> (Include link to Figma designs, if relevant)
[comment]: <> (Include link to Notion ticket, if relevant)

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [ ] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Video of all the screens on an iPad Pro before:
https://user-images.githubusercontent.com/17004429/128683523-611d3f67-b51e-4b42-af6b-50dc8a1a7405.mp4

Video of all the screens on an iPad Pro after:
https://user-images.githubusercontent.com/17004429/128683588-a71d866c-0d33-4f62-a007-0d655c33d13c.mp4

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

File: assets/vaccines/VaccineDemoUK.tsx [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
-const VaccineDemoUK: React.FC<TSvgProps> = () => {
+type TProps = {
+  width?: number;
+};
+
+const defaultWidth = 375;
+const defaultHeight = 262;
+
+export default function VaccineDemoUK(props: TProps) {
+  const ratio = props.width ? props.width / defaultWidth : 1;
```
Comment (selvach) [inline assets/vaccines/VaccineDemoUK.tsx:12]: Could you explain what this is trying to do?

File: src/components/NavHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import * as React from 'react';
+import { StyleSheet, View } from 'react-native';
+
+type TProps = {
+  rightElement?: React.ReactNode;
+};
+
+export function NavHeader(props: TProps) {
+  const navigation = useNavigation();
+  // @todo
```
Comment (selvach) [inline src/components/NavHeader.tsx:13]: Do you know what the TODO is?

File: src/components/Screen.tsx [hunk 3]
```diff
@@ -1,106 +1,103 @@
...
+    </View>
+  ) : null;
 }
 
-export default function Screen(props: TScreenProps) {
-  const scrollEnabled = props.scrollEnabled === undefined ? true : props.scrollEnabled;
+// Keep in mind that certain styling properties don't work on the SafeAreaView.
+// For example setting a padding is ignored.
 
+export function Screen(props: IProps) {
```
Comment (selvach) [inline src/components/Screen.tsx:91]: Lovely refactor!

File: src/features/assessment/TreatmentSelectionScreen.tsx [hunk 4]
```diff
@@ -52,59 +52,40 @@ function TreatmentSelectionScreen({ route }: IProps) {
     >
```
Comment (selvach) [inline src/features/assessment/TreatmentSelectionScreen.tsx:52]: Nice :)

File: src/features/dashboard/CollapsibleHeaderScrollView.tsx [hunk 5]
```diff
@@ -126,7 +126,7 @@ const styles = StyleSheet.create({
   },
   drawerToggleContainer: {
     marginRight: 16,
-    marginTop: 32,
+    marginTop: 32 + 20 / 2,
```
Comment (selvach) [inline src/features/dashboard/CollapsibleHeaderScrollView.tsx:129]: 42?

File: src/features/diet-study-playback/v2/DietStudyGlobalScreen.tsx [hunk 6]
```diff
@@ -1,50 +1,41 @@
...
+    </Screen>
   );
 }
 
 const styles = StyleSheet.create({
   img: {
     aspectRatio: 1,
+    borderRadius: 16,
     height: undefined,
-    resizeMode: 'contain',
```
Comment (selvach) [inline src/features/diet-study-playback/v2/DietStudyGlobalScreen.tsx:47]: Hmm.. I do think contain is better fit. But we can always look at this in QA. 
https://mehrankhandev.medium.com/understanding-resizemode-in-react-native-dd0e455ce63

File: src/features/multi-profile/edit-profile/EditLocationScreen.tsx [hunk 7]
```diff
@@ -126,60 +124,58 @@ export const EditLocationScreen: React.FC<TProps> = (props) => {
...
+              ) : null}
+              {props.values.stillInUK === 'yes' && props.values.differentAddress === 'yes' ? (
                 <GenericTextField
                   required
                   showError
                   formikProps={props}
                   inputProps={{ autoCompleteType: 'postal-code' }}
-                  label={i18n.t('edit-profile.location.label')}
-                  name="postcode"
+                  label={i18n.t('edit-profile.location.other-postcode')}
```
Comment (selvach) [inline src/features/multi-profile/edit-profile/EditLocationScreen.tsx:156]: Nice catch!

File: src/routes/index.tsx [hunk 8]
```diff
@@ -102,7 +103,7 @@ export default function MainNavigator() {
       {ReconsentNavigator({ Stack })}
 
       {/* Uncomment this line to replace the Dashboard with DebugScreens to allow quick debugging */}
-      {/* <Stack.Screen component={DebugScreens} name="Dashboard" options={noHeader} /> */}
+      <Stack.Screen component={DebugScreens} name="Dashboard" options={noHeader} />
```
Comment (selvach) [inline src/routes/index.tsx:106]: Do not check in :)

File: assets/vaccines/VaccineDemoUK.tsx (hunk 1 above)
Comment (aikewoody) [reply assets/vaccines/VaccineDemoUK.tsx:12]: It's used for scaling, on the screen `VaccineFindInfoScreen` we want to set the width of the svg to a specific value and given that number the height needs to change accordingly.

**Review by selvach — CHANGES_REQUESTED** [review]
Great refactor @aikewoody!

File: src/components/NavHeader.tsx (hunk 2 above)
Comment (aikewoody) [reply src/components/NavHeader.tsx:13]: Good spot, it's taken care of

File: src/features/dashboard/CollapsibleHeaderScrollView.tsx (hunk 5 above)
Comment (aikewoody) [reply src/features/dashboard/CollapsibleHeaderScrollView.tsx:129]: Haha yea I think this magic number needs some explanation. I actually changed the code into:
`marginTop: 32 + sizes.drawerToggle / 2,`

Do you think that explains it enough?

Comment (selvach) [reply src/features/dashboard/CollapsibleHeaderScrollView.tsx:129]: I think it does! 👍

File: src/features/diet-study-playback/v2/DietStudyGlobalScreen.tsx (hunk 6 above)
Comment (aikewoody) [reply src/features/diet-study-playback/v2/DietStudyGlobalScreen.tsx:47]: Agreed, to solve this I spent too much time on writing a scalable image but it's very handy!

https://github.com/zoe/covid-tracker-react-native/blob/feature/ipad-screens/src/components/ScalableImage.tsx

**Review by selvach — DISMISSED** [review]
Fantastic!

**Review by bionara — APPROVED** [review]

**Pull request merged**